### ISSUE TITLE
MAX - We should be able to put multiple copies of one collectible in the same container, or across multiple containers

### ISSUE THREAD

**Issue by fearlesswee** [TheGameCreators/GameGuruRepo#4809]
Right now, the "item container" variable appears to be globally shared across all instances of that collectible type, so it is impossible to for example, create a "MedKit" item, and then have multiple containers around the level that might contain a MedKit. It's also impossible to have multiple MedKits inside of a single container. 

This is highly restrictive; the "solution" is to create new MedKit types for each and every instance of a MedKit found throughout your game, which would be extremely confusing to a player who will loot "MedKit278" and "MedKit173" and wonder what the differences between them are.

The most simple solution to this problem would be to make the "item container" variable per-entity, and not shared globally/tied to the "collectible" entry of that item. This would mean you could on a per-item basis designate what container they belong to, and if multiple items have the same container, then that container would have multiple instances of that item type when looted.

**Comment by LeeBamberTGC** [comment]
@fearlesswee Thanks for the enhancement idea. For clarity, the end result would be the ability to create a new item called MediKit, that can be collected as a resource (stacking) so you can collect say 15 medikits in your inventory, and then have the ability to set up multiple containers in your level, and seed medikits in as many containers as you wish, so the player can discover them gradually as they play through your level.

**Comment by fearlesswee** [comment]
I think there's a bit of confusion, I don't mean allowing MedKits to stack, but rather than there can be multiple instances of a MedKit in a container or across multiple containers, stackable or not. Currently, since the "container" field is globally tied to the "collectible" properties of an item and not specific to that instance of the item, this is not possible.

**Comment by fearlesswee** [comment]
I'll use the game "MineCraft" to set up a little example to demonstrate what I mean.

Here are two separate chests.
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/56294129/2e64c0c6-c03d-48db-8d6d-6ef16d8b61bb)

If I open the one on the left, there are two identical copies of a "Healing Potion".
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/56294129/78b2fbbf-7292-4bfa-90d1-eb917f849de4)

If I open the one on the right, it has an identical copy of a "Healing Potion", the very same item in the other chest.
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/56294129/09717f88-378e-4d9b-91d4-4617d6703f91)

Currently, this basic setup is impossible to accomplish in MAX as a set of lootable containers placed by the developer.

**Comment by LeeBamberTGC** [comment]
@fearlesswee If I understand your requirement, it sounds like you want multiple containers with different contents, AND THEN add your collectable to both, and in the quantities you desire, correct?

**Comment by fearlesswee** [comment]
@LeeBamberTGC What I want is to be able to create a collectible item, such as a MedKit, and then, per-instance/copy of that item, designate what container that specific MedKit might belong to. So a player can loot a container near the start of the level and find say, 2 MedKits within it, and then later loot another container later which also has a MedKit in it, maybe then loot yet another container that has 25 9mm rounds and a MedKit, etc. Right now, **ALL** MedKits are forced to share a single container, and only one of them will actually show up, regardless of how many of them you place down, since the "container" field is tied to the collectible entry of the item and not the actual instance/copy of that item. 

This means across my entire game, the player can loot a grand total of **1** singular MedKit from **1** singular container, which is extremely restrictive and counter-intuitive.

**Comment by fearlesswee** [comment]
As an example of what this problem looks like in MAX, here's a chest I set up, and I placed down 4 MedKits, each with their collectible entry stating that this chest is their container. Logic would dictate there should now be 4 MedKits in the chest, right?

![image](https://github.com/TheGameCreators/GameGuruRepo/assets/56294129/25aeea76-e5ac-4474-a741-8de4b102b144)

Unfortunately no, there's only **1** singular MedKit within the chest, despite there being 4 different copies of a MedKit all pointing to this chest as their container.

![image](https://github.com/TheGameCreators/GameGuruRepo/assets/56294129/666a728a-9dec-41a5-988e-3a1248699fa8)

This is because, for whatever reason, the "container" field is a global value tied to the collectible settings of the item, rather than being item-specific. This means that the chest is basically seeing that there is 1 collectible entry for a "MedKit", and that it belongs to this container, therefore there is 1 MedKit in this container. If I were to then try to place more containers each with their own MedKits inside of it, changing the "container" variable would instead now make it so this one singular MedKit instance appears in that chest instead, since the "container" variable is globally tied to the collectible entry, and not the specific entity/copy/instance of that MedKit in the level.

**Comment by LeeBamberTGC** [comment]
@fearlesswee I agree the current system on reflection is a little counter intuitive.  As you can see, to add the pot to the chest you specify the chest when describing the pot:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/6e354a0c-a9f2-447c-9cf5-fdea3fa53d33)

Not very helpful.  The chest on the other hand can be selected but does not really show what is in it, only a reference to the container name "chest":
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/2eea5322-1c24-42de-a98b-a1cbabc57888)

If we looked at this fresh, most users would just want to select a chest (or many chests), then start adding collectables into them with various quantities, like shelf filling.  I feel this is best done in the level editor and in a visible way, so at any time you can select a chest in your game and see what it contains, and drag stuff in and out as fits your design and needs.

Sound better?

**Comment by LeeBamberTGC** [comment]
NOTE: A Simple Way:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/c7b82683-31ad-4e22-836d-1a9f7e6ad564)
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/a766a437-1b11-4e08-8a43-f21443cfe613)

Feel free to comment and feedback, but this seems the quickest and most natural way to populate the chest with objects

**Comment by fearlesswee** [comment]
That definitely sounds very intuitive from an average end-user perspective, and would allow for functionality like "multiple items per container" and "multiple instances of one item type across multiple containers" like I was asking for. I say this is definitely a great solution, but it *might* be worth asking around just to make sure this works for everyone :)

Thank you so much for being so quick to respond and willing to engage with me on this!
